### PR TITLE
[ECO-2117] Add local docker

### PR DESCRIPTION
**Pull request by CRBl69**
# Description

Add a docker compose file to locally deploy the whole stack.

# Testing

Run `docker compose -f compose.yaml -f compose.local.yaml up -d`.

Don't forget to copy `example.local.env` to `.env` first, or alternatively you can run `source example.local.env`.

# Checklist

- [x] Did you check all checkboxes from the linked Linear task?

### CODE REVIEW

**Review by alnoki — CHANGES_REQUESTED** [review]

File: src/docker/compose.local.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      aptos:
+        condition: service_healthy
+
+  aptos:
+    image: aptoslabs/tools:nightly
+    network_mode: host
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
+      - /tmp/testnet:/testnet
+    command: aptos node run-local-testnet --with-indexer-api --with-faucet --force-restart --assume-yes --bind-to 0.0.0.0
```
[the remark this answers is not in the document]
Comment (xbtmatt) [reply src/docker/compose.local.yaml:21]: Removed in #212

File: src/docker/frontend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ENV INBOX_URL=$INBOX_URL
+ENV HASH_SEED=$HASH_SEED
+ENV NEXT_PUBLIC_APTOS_NETWORK=$NEXT_PUBLIC_APTOS_NETWORK
+ENV NEXT_PUBLIC_INTEGRATOR_ADDRESS=$NEXT_PUBLIC_INTEGRATOR_ADDRESS
+ENV NEXT_PUBLIC_INTEGRATOR_FEE_RATE_BPS=$NEXT_PUBLIC_INTEGRATOR_FEE_RATE_BPS
+ENV NEXT_PUBLIC_IS_ALLOWLIST_ENABLED=$NEXT_PUBLIC_IS_ALLOWLIST_ENABLED
+ENV NEXT_PUBLIC_MODULE_ADDRESS=$NEXT_PUBLIC_MODULE_ADDRESS
+ENV NEXT_PUBLIC_MQTT_URL=$NEXT_PUBLIC_MQTT_URL
+ENV NEXT_PUBLIC_REWARDS_MODULE_ADDRESS=$NEXT_PUBLIC_REWARDS_MODULE_ADDRESS
+ENV REVALIDATION_TIME=$REVALIDATION_TIME
```
[the remark this answers is not in the document]
Comment (xbtmatt) [reply src/docker/frontend/Dockerfile:32]: I don't think it does- you need to declare it as an arg for it to be able to be parsed as a variable. At least, I tried it with just `ENV ...` and it didn't work

File: src/docker/example.local.env [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#
+# The same address formating rules apply as above.
+#
+# Emojicoin latest rewards testnet deployment:
+# - 0x22227920701e36651a6649be2067cd7eebf3fabb94717ff3b256e3ada58b2222
+#
+# By default, this is set for the local deployment.
+#
+# Default value: 0x4bab58978ec1b1bef032eeb285ad47a6a9b997d646c19b598c35f46b26ff9ece
+EMOJICOIN_REWARDS_MODULE_ADDRESS="0x4bab58978ec1b1bef032eeb285ad47a6a9b997d646c19b598c35f46b26ff9ece"
```
[the remark this answers is not in the document]
Comment (alnoki) [reply src/docker/example.local.env]: Leaving as same in local for ease of setup

File: src/docker/example.local.env [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+#
+# The same address formating rules apply as above.
+#
+# Emojicoin latest rewards testnet deployment:
+# - 0x33332c9ea4c220e0572b7f83f397164f8171e1c9f681136bb8ab78efa6c43333
+#
+# By default, this is set for the local deployment.
+#
+# Default value: 0x4bab58978ec1b1bef032eeb285ad47a6a9b997d646c19b598c35f46b26ff9ece
+EMOJICOIN_INTEGRATOR_ADDRESS=0x4bab58978ec1b1bef032eeb285ad47a6a9b997d646c19b598c35f46b26ff9ece
```
[the remark this answers is not in the document]
Comment (alnoki) [reply src/docker/example.local.env]: Leaving as same in local for ease of setup

File: src/docker/example.testnet.env [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+#
+# Default value: 3009
+BROKER_PORT="3009"
+
+# The compromised public key for local depolyment.
+#
+# This only needs to be set in case of a local deployment.
+#
+# Default value: 0x29479e9e5fe47ba9a8af509dd6da1f907510bcf8917bfb19b7079d8c63c0b720
+# PUBLISHER_PK="0x29479e9e5fe47ba9a8af509dd6da1f907510bcf8917bfb19b7079d8c63c0b720"
```
[the remark this answers is not in the document]
Comment (alnoki) [reply src/docker/example.testnet.env]: Taken out from testnet env

**Review by alnoki — APPROVED** [review]

**Review by xbtmatt — APPROVED** [review]

**Pull request merged**